### ISSUE TITLE
prefer_is_not: check even if statement contains no "is"

### ISSUE THREAD

**Issue by mbtools** [abaplint/abaplint#1389]
False positive:
```
REPORT zfoobar.

IF NOT 1 = 2.
  WRITE 1.
ENDIF.
```

/zfoobar.prog.abap[3, 4]: Prefer IS NOT to NOT IS(prefer_is_not)

The error should only appear is "IS" is part of the statement.

**Comment by larshp** [comment]
https://github.com/SAP/styleguides/blob/master/clean-abap/CleanABAP.md#prefer-is-not-to-not-is

**Comment by larshp** [comment]
hmm, one of the anti patterns given by the style guide is the below, which corresponds to the example and finding

```abap
" anti-pattern
IF NOT variable = 42.
```

**Comment by mbtools** [comment]
Got it. The examples are clear. The rule name/description confused me a bit
